merge/update.go

```diff
@@ -200,11 +200,11 @@ func (s *Updater) Apply(liveObject, configObject *typed.TypedValue, version fiel
 	if err != nil {
 		return nil, fieldpath.ManagedFields{}, fmt.Errorf("failed to prune fields: %v", err)
 	}
-	managers, compare, err := s.update(liveObject, newObject, version, managers, manager, force)
+	managers, _, err = s.update(liveObject, newObject, version, managers, manager, force)
 	if err != nil {
 		return nil, fieldpath.ManagedFields{}, err
 	}
-	if compare.IsSame() {
+	if value.Equals(liveObject.AsValue(), newObject.AsValue()) {
```

Comment:
is this expensive

Reply:
Yes, benchmarks at the top ... :-(

Reply:
Not sure why I'm not using the allocator here...